cloud/passportauth/integration/UserDataSource.ts

```diff
@@ -0,0 +1,26 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import {User} from './UserSeedData';
+
+interface DataSource<T> {
+  read(id: string): object;
+}
+
+export class UserDataSource<T extends User> implements DataSource<T> {
```

Comment:
I will go with DataRepository as we using this pattern across the Raincatcher.
"DataSource" - Represents unstructurized DataSource
"DataRepository" - Pattern to define set of the functions that will return expected results.